packages/@lwc/wire-service/README.md

```diff
@@ -1,164 +1,74 @@
-interface ValueChangedEvent {
-    value: any;
-    new(value: any) : ValueChangedEvent;
-}
-
-// Event types the wire adapter may listen for
-type EventType = 'config' | 'connect' | 'disconnect';
+// bookItem.js
+import { LightningElement } from 'lwc';
 
-// Event listener callback
-type Listener = (config?: { [key: string]: any ) => void;
+export default class WireExample extends {
+    @api bookId;
 
-// Target of the @wire
-interface WireEventTarget extends EventTarget {
-    dispatchEvent(event: ValueChangedEvent): boolean;
-    addEventListener(type: EventType, listener: Listener): void;
-    removeEventListener(type: EventType, listener: Listener): void;
+    @wire(getBook, { id: '$bookId'})
+    book;
 }
 ```
 
-In the component's `wiring` lifecycle, the wire service invokes the `wireAdapterFactory` function to configure an instance of the wire adapter for each `@wire` instance (which is per component instance).
-
-`eventTarget` is an implementation of [Event Target](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) that supports listeners for the following events:
-
--   `config` is delivered when the resolved configuration changes. A singular argument is provided: the resolved configuration.
--   `connect` is delivered when the component is connected.
--   `disconnect` is delivered when the component is disconnected.
-
-The wire service remains responsible for resolving the configuration object. `eventTarget` delivers a `config` event when the resolved configuration changes. The value of the configuration is specific to the wire adapter. The wire adapter must treat the object as immutable.
-
-The wire adapter is responsible for provisioning values by dispatching a `ValueChangedEvent` to the event target. `ValueChangedEvent`'s constructor accepts a single argument: the value to provision. There is no limitation to the shape or contents of the value to provision. The event target handles property assignment or method invocation based on the target of the `@wire`.
-
-The wire adapter is responsible for maintaining any context it requires. For example, tracking the values it provisions and the originating resolved configuration is shown in the basic example below.
-
-### Registering A Wire Adapter
-
-A wire adapter must be registered with the wire service. The `wire-service` module provides a registration function.
-
-```ts
-register(adapterId: WireAdapterId, wireAdapterFactory: WireAdapterFactory);
-```
-
-## Example Wire Adapter Implementation
+Wire adapters are part of LWC's reactivity system. An `@wire` takes the name of a wire adapter and an optional configuration object, which is specific to the wire adapter. You can use a `$` to mark the property of a configuration object as reactive. When a reactive property’s value changes, the wire adapter's `update` method executes with the new value. When the wire adapter provisions new data, the component rerenders.
```

Comment:
Rerendering is controlled by LWC's regular rerendering logic: has a value used in the template changed.

```suggestion
Wire adapters are part of LWC's reactivity system. An `@wire` takes the name of a wire adapter and an optional configuration object, which is specific to the wire adapter. You can use a `$` to mark the property of a configuration object as reactive. When a reactive property’s value changes, the wire adapter's `update` method executes with the new value. When the wire adapter provisions new data, the component rerenders if necessary.
```